File: controller-web/src/main/java/top/emilejones/hhu/web/config/BeanConfiguration.java
package top.emilejones.hhu.web.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Import;
import top.emilejones.hhu.milvus.SingleCollectionMilvusRepository;
import top.emilejones.hhu.model.ModelClient;
import top.emilejones.hhu.model.impl.ModelClientByHttp;
import top.emilejones.hhu.repository.impl.neo4j.Neo4jRepositoryImpl;
import top.emilejones.hhu.repository.IMilvusRepository;
import top.emilejones.hhu.repository.INeo4jRepository;
import top.emilejones.hhu.service.impl.RecallService;
import top.emilejones.huu.env.AutoFindConfigFile;
import top.emilejones.huu.env.pojo.HttpModelClientConfig;
import top.emilejones.huu.env.pojo.MilvusConfig;
import top.emilejones.huu.env.pojo.Neo4jConfig;
import top.emilejones.huu.env.pojo.RAGConfig;

@Configuration
@Import(AutoFindConfigFile.class)
public class BeanConfiguration {
    @Bean
    public ModelClient getModelClient(HttpModelClientConfig modelConfig) {
        return new ModelClientByHttp(
                modelConfig.getHost(),
                modelConfig.getPort(),
                modelConfig.getToken(),
                modelConfig.getEmbeddingModel(),
                modelConfig.getRerankModel()
        );
    }

    @Bean
    public RecallService getRecallService(RAGConfig ragConfig, IMilvusRepository milvusRepository, INeo4jRepository neo4jRepository, ModelClient modelClient) {
        return new RecallService(milvusRepository, neo4jRepository, modelClient, ragConfig.getRecallNumber());
    }

    @Bean
    public IMilvusRepository getMilvusRepository(MilvusConfig milvusConfig, HttpModelClientConfig modelConfig) {
        return new SingleCollectionMilvusRepository(milvusConfig.getHost(),
                milvusConfig.getPort(),
                milvusConfig.getDatabase(),
                milvusConfig.getCollection(),
                modelConfig.getDimension());
    }

    @Bean
    public INeo4jRepository getNeo4jRepository(Neo4jConfig neo4jConfig) {
        return new Neo4jRepositoryImpl(neo4jConfig.getHost(), neo4jConfig.getPort(), neo4jConfig.getUser(), neo4jConfig.getPassword(), neo4jConfig.getDatabase());
    }
}

File: infra-textsplitter/src/main/kotlin/top/emilejones/hhu/textsplitter/config/BeanConfiguration.kt
package top.emilejones.hhu.textsplitter.config

import io.milvus.v2.client.ConnectConfig
import io.milvus.v2.client.MilvusClientV2
import org.neo4j.driver.AuthTokens
import org.neo4j.driver.Driver
import org.neo4j.driver.GraphDatabase
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import top.emilejones.hhu.env.pojo.MilvusConfig
import top.emilejones.hhu.env.pojo.Neo4jConfig

@Configuration
class BeanConfiguration {

    @Bean
    fun neo4jDriver(neo4jConfig: Neo4jConfig): Driver {
        return GraphDatabase.driver(
            "bolt://${neo4jConfig.host}:${neo4jConfig.port}",
            AuthTokens.basic(neo4jConfig.user, neo4jConfig.password)
        )
    }

    @Bean
    fun milvusClient(milvusConfig: MilvusConfig): MilvusClientV2 {
        return MilvusClientV2(
            ConnectConfig.builder()
                .uri("http://%s:%d".format(milvusConfig.host, milvusConfig.port))
                .build()
        )
    }
}